integration-tests/src/main/java/eu/tsystems/mms/tic/testframework/core/utils/Log4jFileReader.java

```diff
@@ -0,0 +1,166 @@
+        return filteredLogEntries;
+    }
+
+    /**
+     * @param searchString
+     * @return
+     */
+    public List<String> filterLogForString(final String searchString) {
+        final List<String> allLogEntries = readLines();
+        if (searchString != null) {
+            List<String> filteredLogEntries = allLogEntries.stream().filter(line -> line.contains(searchString)).collect(Collectors.toList());
+            log().info(String.format("Found %s log entries for '%s'.", filteredLogEntries.size(), searchString));
+            log().info("Found entries: " + Arrays.toString(filteredLogEntries.toArray(new String[0])));
+            return filteredLogEntries;
+        }
+        return Collections.EMPTY_LIST;
+    }
+
+    /**
+     * find an entry existing in the log directly after given searchString
+     *
+     * @param searchString entry to find first following entry
+     * @return String following entry
+     */
+    public String filterLogForFollowingEntry(final String searchString) {
+        final List<String> allLogEntries = readLines();
+        String matchedEntry = "not_found";
+
+        if (searchString != null) {
+            for (int i = 0; i < allLogEntries.size(); i++) {
+                final String foundEntry = allLogEntries.get(i);
+
+                if (foundEntry.equals(searchString)) {
+                    log().info(String.format("Found ParentEntry: '%s'. ", searchString));
+
+                    matchedEntry = allLogEntries.get(i + 1);
+                    log().info(String.format("Found ChildEntry '%s'. ", matchedEntry));
+                    break;
+                }
+            }
+        }
+
+        log().info(String.format("Returning entry '%s'.", matchedEntry));
+        return matchedEntry;
+    }
+
+    /**
+     * verify expectedStatus for all entries found for classNameSlug and methodNameSlug
+     *
+     * @param classNameSlug className in log file entry
+     * @param methodNameSlug methodName in log file entry
+     * @param expectedStatus expectedStatus for found entries
+     */
+    public void assertTestStatusPerMethod(final String classNameSlug, final String methodNameSlug, final Status expectedStatus) {
+        final List<String> foundEntries = filterLogForTestMethod(classNameSlug, methodNameSlug);
+
+        for (final String entry : foundEntries) {
+            log().debug(String.format("Asserting '%s' for '%s'",
+                    expectedStatus, methodNameSlug));
+            Assert.assertTrue(entry.contains(expectedStatus.title),
+                    String.format("'%s' has status '%s'", methodNameSlug, expectedStatus));
+        }
+    }
+
+    /**
+     * read log line by line
+     *
+     * @return List<String> of found entries
+     */
+    private List<String> readLines() {
+
+        final File logFile = Paths.get(PATH_TO_LOG_FILE).toFile();
+        try {
+            return new LinkedList<>(org.apache.commons.io.FileUtils.readLines(logFile, Charset.defaultCharset()));
```

Comment:
Mmh. Just as a heads-up, `Charset.defaultCharset()` returns the default charset _of the JVM_ . Not necessarily of the OS or any file. It will probably always work, but if we do use this method, we should really ensure we have a default charset _set_! The default charset is inherited from the OS at JVM start, so it could be something horrible like `cp1252` and _could be different for different installations_ (notably on Windows vs. Linux hosts).